server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -0,0 +1,251 @@
+            if (usage == null) {
+                return new IndividualNodeHealth.Disk(HealthStatus.UNKNOWN, IndividualNodeHealth.Disk.Cause.NODE_HAS_NO_DISK_STATS);
+            }
+
+            ByteSizeValue totalBytes = ByteSizeValue.ofBytes(usage.getTotalBytes());
+
+            if (node.isDedicatedFrozenNode()) {
+                long frozenFloodStageThreshold = diskMetadata.getFreeBytesFrozenFloodStageWatermark(totalBytes).getBytes();
+                if (usage.getFreeBytes() < frozenFloodStageThreshold) {
+                    logger.debug("flood stage disk watermark [{}] exceeded on {}", frozenFloodStageThreshold, usage);
+                    return new IndividualNodeHealth.Disk(
+                        HealthStatus.RED,
+                        IndividualNodeHealth.Disk.Cause.FROZEN_NODE_OVER_FLOOD_STAGE_THRESHOLD
+                    );
+                }
+                return new IndividualNodeHealth.Disk(HealthStatus.GREEN);
+            }
+
+            long floodStageThreshold = diskMetadata.getFreeBytesFloodStageWatermark(totalBytes).getBytes();
+            if (usage.getFreeBytes() < floodStageThreshold) {
+                return new IndividualNodeHealth.Disk(HealthStatus.RED, IndividualNodeHealth.Disk.Cause.NODE_OVER_THE_FLOOD_STAGE_THRESHOLD);
+            }
+
+            long highThreshold = diskMetadata.getFreeBytesHighWatermark(totalBytes).getBytes();
+            if (usage.getFreeBytes() < highThreshold && hasRelocatingShards(clusterState, node.getId()) == false) {
+                return new IndividualNodeHealth.Disk(HealthStatus.YELLOW, IndividualNodeHealth.Disk.Cause.NODE_OVER_HIGH_THRESHOLD);
+            }
+            return new IndividualNodeHealth.Disk(HealthStatus.GREEN);
+        }
+
+        private DiskUsage getDiskUsage() {
+            NodeStats nodeStats = nodeService.stats(
+                CommonStatsFlags.NONE,
+                false,
+                false,
+                false,
+                false,
+                true,
+                false,
+                false,
+                false,
+                false,
+                false,
+                false,
+                false,
+                false,
+                false
+            );
+            final String nodeId = nodeStats.getNode().getId();
+            final String nodeName = nodeStats.getNode().getName();
+            if (nodeStats.getFs() == null) {
+                logger.debug("node [{}/{}] did not return any filesystem stats", nodeName, nodeId);
+                return null;
+            }
+
+            FsInfo.Path leastAvailablePath = null;
+            for (FsInfo.Path info : nodeStats.getFs()) {
+                if (leastAvailablePath == null) {
+                    leastAvailablePath = info;
+                } else if (leastAvailablePath.getAvailable().getBytes() > info.getAvailable().getBytes()) {
+                    leastAvailablePath = info;
+                }
+            }
```

Comment:
would we be interested in the path ES is configured to store data? ie. `path.data` ?

Reply:
Hm.... But that wouldn't work for non-data nodes, right?

But even for data nodes, the `DiskThresholdMonitor` also checks for the least available path and I think it would be nice to be consistent. 

What do you think?